app/models/concerns/facets/managed_host_extensions.rb

```diff
@@ -0,0 +1,66 @@
+require 'facets'
+
+module Facets
+  module ManagedHostExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      Facets::ManagedHostExtensions.refresh_facet_relations(self)
+
+      def attributes
+        hash = super
+
+        # include all facet attributes by default
+        facets_with_definitions.each do |facet, facet_definition|
+          hash["#{facet_definition.name}_attributes"] = facet.attributes.reject { |key, _| %w(created_at updated_at).include? key }
+        end
+        hash
+      end
+    end
+
+    def self.refresh_facet_relations(klass)
+      Facets.registered_facets.values.each do |facet_config|
+        self.register_facet_relation(klass, facet_config)
+      end
+    end
+
+    def self.register_facet_relation(klass, facet_config)
+      klass.class_eval do
+        has_one facet_config.name, :class_name => facet_config.model.name, :foreign_key => :host_id, :inverse_of => :host
+        accepts_nested_attributes_for facet_config.name
+        alias_method "#{facet_config.name}_attributes", facet_config.name
+
+        include facet_config.extension if facet_config.extension
+
+        include_in_clone facet_config.name
+
+        facet_config.compatibility_properties.each do |prop|
+          define_method(prop) { |*args| forward_property_call(prop, args, facet_config.name) }
+        end if facet_config.compatibility_properties
+      end
+    end
+
+    def facets
+      facets_with_definitions.keys
+    end
+
+    # This method will return a hash of facets for a specific host including the coresponding definitions.
+    # The output should look like this:
+    # { host.puppet_aspect => Facets.registered_facets[:puppet_aspect] }
+    def facets_with_definitions
```

Comment:
`#facets` returns an array of active records - the actual facet objects, not names.
`#facets_with_definitions` returns tuples: facet record as a key for the hash, same facet's configuration object as a value. Will it be better if the return value would be array of arrays?
Something like this: 
``` Ruby
[[facet1_instance, facet1_config], [facet2_instance, facet2_config]]
```
Or maybe the better solution would be just renaming the method to `#facet_and_cofigurations_hash`?